src/testdrive/src/action.rs

```diff
@@ -120,6 +125,93 @@ impl State {
         }
         Ok(())
     }
+
+    /// Delete S3 buckets that were created in this run
+    pub(crate) async fn reset_s3(&mut self) -> Result<(), Error> {
+        if self.s3_buckets_created.is_empty() {
+            return Ok(());
+        }
+
+        let buckets: Vec<String> = self.s3_buckets_created.iter().cloned().collect();
+
+        let mut errors: Vec<Box<dyn std::error::Error + Send + Sync>> = Vec::new();
+        for bucket in buckets.into_iter() {
+            // you can't delete a bucket that has stuff in it
+            let mut continuation_token = None;
+            loop {
+                let response = self
+                    .s3_client
+                    .list_objects_v2(ListObjectsV2Request {
+                        bucket: bucket.clone(),
+                        continuation_token: continuation_token.take(),
+                        ..Default::default()
+                    })
+                    .await;
+                match response {
+                    Ok(response) => {
+                        if let Some(objects) = response.contents {
+                            for obj in objects {
+                                if let Err(e) = self
+                                    .s3_client
+                                    .delete_object(DeleteObjectRequest {
+                                        bucket: bucket.clone(),
+                                        key: obj.key.clone().unwrap(),
+                                        ..Default::default()
+                                    })
+                                    .await
+                                {
+                                    errors.push(
+                                        Error::with_cause(
+                                            format!(
+                                                "Deleting object {}/{}",
+                                                bucket,
+                                                obj.key.unwrap()
+                                            ),
+                                            e,
+                                        )
+                                        .into(),
+                                    );
+                                }
+                            }
+                        }
+
+                        if response.next_continuation_token.is_none() {
+                            break;
+                        }
+                        continuation_token = response.next_continuation_token;
+                    }
+                    Err(e) => {
+                        errors.push(
+                            Error::with_cause(format!("Listing objects for bucket {}", bucket), e)
+                                .into(),
+                        );
```

Comment:
This doesn't look quite right to me. If you encounter an error, seems like you'll get stuck in the loop forever.

Reply:
Yeah a repeating error would do that, switched this whole thing to use `ore::retry`